cachito/web/api_v1.py

```diff
@@ -588,3 +590,41 @@ def get_request_logs(request_id):
     return flask.Response(
         stream_with_context(generate_stream_response(log_file_path)), mimetype="text/plain"
     )
+
+
+@api_v1.route("/content-manifest", methods=["GET"])
+def get_content_manifest_by_requests():
+    """
+    Retrieve the content manifest associated with the given requests.
+
+    :return: a Flask JSON response
+    :rtype: flask.Response
+    :raise NotFound: if one of the given request is not found
+    :raise BadRequest: if any of the given request is not in the "complete" or "stale" state
+    """
+    arg = flask.request.args.get("requests")
+    if not arg:
+        return flask.jsonify(BASE_ICM)
+    request_ids = []
+    item: str
+    for item in arg.split(","):
+        if not item.strip():
+            continue
+        if not item.strip().isdigit():
+            raise BadRequest(f"{item} is not an integer.")
+        request_ids.append(int(item))
+
+    requests = Request.query.filter(Request.id.in_(request_ids))
```

Comment:
This will ignore missing requests, but the docstring says that missing requests will cause a `NotFound` error (and it should raise the error IMO).